qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -242,10 +230,25 @@ class SymbolicPulse(Pulse):
     concatenated with logical expressions such as ``And`` and ``Or`` in SymPy or Symengine.
     The symbolic pulse instance can be played only when the constraint function returns ``True``.
     The constraint is evaluated when :meth:`SymbolicPulse.validate_parameters` is called.
-    Note that the maximum pulse amplitude limit is separately evaluated when
-    the :attr:`.limit_amplitude` is set.
-    Since this is evaluated with actual waveform samples by calling :meth:`.get_waveform`,
-    it is not necessary to define any explicit constraint for the amplitude limitation.
+
+
+    .. _symbolic_pulse_eval_condition:
+
+    .. rubric:: Maximum amplitude validation
```

Comment:
Probably I was writing the code too specific to IBM hardware backends. I lean toward implementing a base class more flexibly. In this sense, it was tough to write a clean logic that goes straight to the error, without assuming a special parameter. I think checking full waveform is practically less expensive, since most of user-code doesn't intentionally hit this condition. 

I like `valid_amp_conditions`. This is more intuitive. Done in https://github.com/Qiskit/qiskit-terra/pull/7821/commits/045d67bee4efdc36ac6e2d3bbd991737d1eca99c